File: sorting-algos/gnome_sort/gnome_sort.c
#include <stdio.h>

void input(int arr[],int size)
{
    int i;

    printf("Enter the array elements\n");

    for(i=0;i<size;i++)

        scanf("%d",&arr[i]);
}

void print_array(int arr[],int size)
{
    printf("The sorted array is\n");
    int i;
    for(i=0;i<size;i++)
        printf("%d ",arr[i]);
    printf("\n");
}
void gnome_sort(int arr[],int size)
{
    int i=0,temp;

    while(i<size)
    {
        if(i==0)
            i++;    //as first element has no previous element

        else if(arr[i-1]>arr[i])    //checks if previous element is larger then swaps and i is decreased by 1
        {
            temp=arr[i];
            arr[i]=arr[i-1];
            arr[i-1]=temp;
            i--;
        }
        else
            i++;    //moves i forward
    }
}


int main()
{
    printf("Enter the number of elements in array\n");

    int size,i;

    scanf("%d",&size);

    int arr[size];

    input(arr,size);    //for getting input array

    gnome_sort(arr,size); //sorts using gnome sort

    print_array(arr,size);//prints the sorted array
}

File: sorting-algos/gnome_sort/gnome_sort.py
def gnome_sort(arr,size):
    i=0
    while(i<size):
        if (i==0):          #first element has no previous element
            i=i+1
        elif(arr[i-1]>arr[i]):
            temp=arr[i-1]
            arr[i-1]=arr[i]
            arr[i]=temp
            i=i-1           #swaps previous element with current and moves one step back
        else:
            i=i+1           #moves one step front


if(__name__=='__main__'):
    size=int(input('Enter the number of elements in array\n'))
    print('Enter the elements')
    arr=[int(input()) for i in range(size)]
    gnome_sort(arr,size)
    print('The sorted array is')
    print(arr)